File: app/src/main/java/com/artamonov/millionplanets/sectors/SectorsYouAdapter.java
package com.artamonov.millionplanets.sectors;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.NumberPicker;
import android.widget.TextView;
import android.widget.Toast;

import com.artamonov.millionplanets.R;
import com.artamonov.millionplanets.model.ObjectModel;
import com.artamonov.millionplanets.model.User;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class SectorsYouAdapter extends RecyclerView.Adapter<SectorsYouAdapter.ViewHolder> {

    //private static ItemClickListener listener;
    private static DialogListener dialogListener;
    private static List<User> userList;
    private static Context context;
    private List<ObjectModel> objectModelList;
    private NumberPicker numberPicker;
    private boolean isPlanetTab;

    public SectorsYouAdapter(List<User> userList, List<ObjectModel> objectModelList, Context context, DialogListener listener) {
        this.userList = userList;
        dialogListener = listener;
        this.objectModelList = objectModelList;
        this.context = context;
        // this.isPlanetTab = isPlanetTab;

    }

    public SectorsYouAdapter(List<User> userList, Context context, SectorsYouAdapter.DialogListener listener) {
        this.userList = userList;
        this.context = context;
        dialogListener = listener;
    }


    @NonNull
    @Override
    public SectorsYouAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.sectors_you_items, parent, false);
        Log.i("myTags", "onCreateViewHolder " + userList.get(0).getResource_iron());
        return new SectorsYouAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SectorsYouAdapter.ViewHolder holder, int position) {
        Log.i("myTags", "onBindViewHolder pos: " + position);
        User user = userList.get(position);
        ObjectModel objectModel = objectModelList.get(position);
        Log.i("myTags", "onBindViewHolder res name: " + objectModel.getResourceName());
        Log.i("myTags", "onBindViewHolder price buy : " + objectModel.getPrice_buy_iron());

        holder.resourcePrice.setText(Integer.toString(objectModel.getPlanetSectorsPrice() / 2));
        holder.resourceAmount.setText(Integer.toString(user.getSectors()));
    }


    @Override
    public int getItemCount() {
        Log.i("myTags", "getItemCount size: " + userList.size());
        return userList.size();
    }


    public interface DialogListener {
        void onDialogCreate();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private final TextView resourceAmount;
        private final TextView resourcePrice;

        ViewHolder(View itemView) {
            super(itemView);
            resourceAmount = itemView.findViewById(R.id.sectors_resource_amount);
            resourcePrice = itemView.findViewById(R.id.sectors_resource_price);
            //  itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (userList.get(getAdapterPosition()).getResource_iron() == 0) {
                Toast.makeText(context, "No resources to sell", Toast.LENGTH_SHORT).show();
                return;
            }
            dialogListener.onDialogCreate();

        }
    }
}

File: app/src/main/java/com/artamonov/millionplanets/sectors/SectorsYouAdapter.kt
package com.artamonov.millionplanets.sectors

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.artamonov.millionplanets.R
import com.artamonov.millionplanets.model.SpaceObject
import com.artamonov.millionplanets.model.User
import kotlinx.android.synthetic.main.market_you_items.view.*
import kotlinx.android.synthetic.main.sectors_you_items.view.*

class SectorsYouAdapter(
    val userList: List<User>,
    private var spaceObjectList: List<SpaceObject>,
    val context: Context,
    val listener: DialogListener
) : RecyclerView.Adapter<SectorsYouAdapter.ViewHolder>() {

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.sectors_you_items, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindItem()
    }

    override fun getItemCount(): Int {
        return userList.size
    }

    interface DialogListener {
        fun onDialogCreate()
    }

    inner class ViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bindItem() {
            itemView.sectors_resource_price.text = (spaceObjectList[adapterPosition].planetSectorsPrice / 2).toString()
            itemView.sectors_resource_amount.text = userList[adapterPosition].sectors.toString()
            itemView.setOnClickListener {
                if (userList[adapterPosition].resource_iron == 0L) {
                    Toast.makeText(context, "No resources to sell", Toast.LENGTH_SHORT).show()
                    return@setOnClickListener
                }
                listener.onDialogCreate()
            }
        }
    }
}